diff --git a/node_modules/@react-navigation/stack/lib/module/views/Stack/CardContainer.js b/node_modules/@react-navigation/stack/lib/module/views/Stack/CardContainer.js
index 28e17dd..ebb6d29 100644
--- a/node_modules/@react-navigation/stack/lib/module/views/Stack/CardContainer.js
+++ b/node_modules/@react-navigation/stack/lib/module/views/Stack/CardContainer.js
@@ -266,7 +266,7 @@ const styles = StyleSheet.create({
     flex: 1
   },
   header: {
-    zIndex: 1
+    zIndex: 1000
   },
   scene: {
     flex: 1
