diff --git a/node_modules/@react-navigation/stack/lib/commonjs/views/Stack/CardContainer.js b/node_modules/@react-navigation/stack/lib/commonjs/views/Stack/CardContainer.js
index 84176e6..cbe3285 100644
--- a/node_modules/@react-navigation/stack/lib/commonjs/views/Stack/CardContainer.js
+++ b/node_modules/@react-navigation/stack/lib/commonjs/views/Stack/CardContainer.js
@@ -273,7 +273,7 @@ const styles = _reactNative.StyleSheet.create({
     flex: 1
   },
   header: {
-    zIndex: 1
+    zIndex: 2000
   },
   scene: {
     flex: 1
