diff --git a/node_modules/react-native-camera-kit/ios/ReactNativeCameraKit/CKCameraViewComponentView.mm b/node_modules/react-native-camera-kit/ios/ReactNativeCameraKit/CKCameraViewComponentView.mm
index 5924c82..7eba869 100644
--- a/node_modules/react-native-camera-kit/ios/ReactNativeCameraKit/CKCameraViewComponentView.mm
+++ b/node_modules/react-native-camera-kit/ios/ReactNativeCameraKit/CKCameraViewComponentView.mm
@@ -180,14 +180,18 @@ - (void)updateProps:(const Props::Shared &)props oldProps:(const Props::Shared &
         _view.ratioOverlayColor = ratioOverlayColor;
         [changedProps addObject:@"ratioOverlayColor"];
     }
-    id scanBarcode = CKConvertFollyDynamicToId(newProps.scanBarcode);
-    if (scanBarcode != nil) {
-        _view.scanBarcode = scanBarcode;
+    // id scanBarcode = CKConvertFollyDynamicToId(newProps.scanBarcode);
+    // if (scanBarcode != nil) {
+    //     _view.scanBarcode = scanBarcode;
+    if (_view.scanBarcode != newProps.scanBarcode) {
+        _view.scanBarcode = newProps.scanBarcode;
         [changedProps addObject:@"scanBarcode"];
     }
-    id showFrame = CKConvertFollyDynamicToId(newProps.showFrame);
-    if (showFrame != nil) {
-        _view.showFrame = showFrame;
+    // id showFrame = CKConvertFollyDynamicToId(newProps.showFrame);
+    // if (showFrame != nil) {
+    //     _view.showFrame = showFrame;
+    if (_view.showFrame != newProps.showFrame) {
+        _view.showFrame = newProps.showFrame;
         [changedProps addObject:@"showFrame"];
     }
     id scanThrottleDelay = CKConvertFollyDynamicToId(newProps.scanThrottleDelay);
@@ -222,7 +226,7 @@ - (void)updateProps:(const Props::Shared &)props oldProps:(const Props::Shared &
     }
     id zoomMode = CKConvertFollyDynamicToId(newProps.zoomMode);
     if (zoomMode != nil) {
-        _view.zoomMode = [focusMode isEqualToString:@"on"] ? CKZoomModeOn : CKZoomModeOff;
+        _view.zoomMode = [zoomMode isEqualToString:@"on"] ? CKZoomModeOn : CKZoomModeOff;
         [changedProps addObject:@"zoomMode"];
     }
     id zoom = CKConvertFollyDynamicToId(newProps.zoom);
